storage/bolt/share.go

```diff
@@ -75,3 +75,11 @@ func (s shareBackend) Delete(hash string) error {
 	}
 	return err
 }
+
+func (s shareBackend) DeleteWithPath(path string) error {
+	err := s.db.Select(q.Eq("Path", path)).Delete(&share.Link{})
+	if errors.Is(err, storm.ErrNotFound) {
+		return nil
+	}
+	return err
+}
```

Comment:
I think there might be a more efficient way to do this. I see the `path` is an index in storm. Isn't it possible to use the `db.Prefix` to delete every matching share with the path as prefix?